cdap-ui/app/cdap/services/redirect-to-login.js

```diff
@@ -17,9 +17,12 @@
 import cookie from 'react-cookie';
 export default function RedirectToLogin(data) {
   let {statusCode} = data;
-  let {url} = data.resource;
+  let {url} = data.resource || {};
```

Comment:
shouldn't this be `data.resource || ''`?

Reply:
No. `resource` is an object with url, header and other properties that we pass to express server.